src/hooks/posters.ts

```diff
@@ -47,37 +49,51 @@ export const usePosters = (posterHallId: string) => {
   } = useDebounceSearch();
 
   const [liveFilter, setLiveFilter] = useState<boolean>(false);
+  const [displayedPostersCount, setDisplayedPostersAmount] = useState(
+    DEFAULT_DISPLAYED_POSTER_PREVIEW_COUNT
+  );
+
+  const increaseDisplayedPosterCount = useCallback(() => {
+    setDisplayedPostersAmount(
+      (prevPostersNumber) =>
+        prevPostersNumber + DEFAULT_DISPLAYED_POSTER_PREVIEW_COUNT
+    );
+  }, []);
+
+  const filteredPosterVenues = useMemo(
+    () =>
+      liveFilter
+        ? posterVenues.filter((posterVenue) => posterVenue.isLive)
+        : posterVenues,
+    [posterVenues, liveFilter]
+  );
 
   const fuseVenues = useMemo(
     () =>
-      new Fuse(posterVenues, {
+      new Fuse(filteredPosterVenues, {
         keys: ["poster.title", "poster.authorName", "poster.categories"],
       }),
-    [posterVenues]
+    [filteredPosterVenues]
   );
 
   const searchedPosterVenues = useMemo(() => {
-    if (!searchQuery) return posterVenues;
+    if (!searchQuery)
+      return filteredPosterVenues.slice(0, displayedPostersCount);
 
     return fuseVenues
       .search(searchQuery)
+      .slice(0, displayedPostersCount)
```

Comment:
I can see what you're doing, using `.slice` to reduce the number of displayed venues, but I expect the crux of the performance issue here is either in the initial loading of the data, or in processing the parts above. I would be very surprised if it was actually at the 'rendering' part after here that everything slowed down.